### PR TITLE
image/spec: add historical information about v1 spec

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

I needed these v1 change logs so as to implement v1 spec on containerd, but could not find any concise document about that.

https://github.com/containerd/containerd/pull/1602

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
@aaronlehmann @dmcgowan @stevvooe

**Review by thaJeztah — APPROVED** [review]
SGTM

**Review by yongtang — APPROVED** [review]
LGTM

**Comment by stevvooe** [comment]
This is not quite right. Its trying to show V1 and V2 as mutually exclusive, when they are different components. The image config, known as 1.2 here, is actually still used in the image format. The format never really reved to V2, only the HTTP API. The manifests and manifest lists are part of the distribution specification.

What is "1.2" became the OCI Image config, while the distribution data structures are the manifest and index components in OCI.

**Comment by AkihiroSuda** [comment]
@stevvooe OK, updated PR

**Comment by ehazlett** [comment]
ping @stevvooe

**Comment by AkihiroSuda** [comment]
ping @stevvooe ptal?

File: image/spec/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+This directory contains documents about Docker Image Specification v1.X.
+
+Distribution and archival specifications such as file layouts and manifests are no longer used
+in Moby and Docker, except in `docker save` and `docker load`.
+
+However, v1 Image JSON (`application/vnd.docker.container.image.v1+json`) has been still widely
+used and officially adopted in [V2 manifest](https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-2.md)
+and in [OCI Image Format Specification](https://github.com/opencontainers/image-spec), which is
+the current industry's standard image specification.
```
Comment (stevvooe) [inline image/spec/README.md]: No need for value statements. Just provide facts.

File: image/spec/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+### [v1.1](v1.1.md)
+
+* Implemented in Docker v1.10 (February, 2016)
+* The official spec document was written in April 2016 ([#22264](https://github.com/moby/moby/pull/22264))
+
+Changes:
+
+* IDs were made into SHA256 digest values rather than random values
+* Layer directory names were made into deterministic values rather than random ID values
```
Comment (stevvooe) [inline image/spec/README.md]: While this impacts the specification, this was a change in the implementation. Should this be included in a historical account of the specification? May have an impact on docker section for each version. For example, 1.9 required migration and 1.10 included these changes.

File: image/spec/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Docker Image Specification v1.
+
+This directory contains documents about Docker Image Specification v1.X.
+
+Distribution and archival specifications such as file layouts and manifests are no longer used
```
Comment (stevvooe) [inline image/spec/README.md]: This statement is odd. Distribution specifications are 100% in use in docker today.

File: image/spec/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+### [v1](v1.md)
+
+* The initial revision
+* The official spec document was written in late 2014 ([#9560](https://github.com/moby/moby/pull/9560)), but actual implementations had existed even earlier
+
+
+## Related specifications
+
+* [Open Containers Initiative (OCI) Image Format Specification v1.0.0](https://github.com/opencontainers/image-spec/tree/v1.0.0)
```
Comment (stevvooe) [inline image/spec/README.md]: Include the Registry API specification here, as well.

File: image/spec/README.md (hunk 1 above)
Comment (AkihiroSuda) [reply image/spec/README.md]: done

File: image/spec/README.md (hunk 3 above)
Comment (AkihiroSuda) [reply image/spec/README.md]: done

File: image/spec/README.md (hunk 4 above)
Comment (AkihiroSuda) [reply image/spec/README.md]: done

File: image/spec/README.md (hunk 2 above)
Comment (AkihiroSuda) [reply image/spec/README.md]: Could you elaborate what change is needed?
Do you mean v1.9 made names deterministic but did not introduce the new manifest JSON?

**Comment by AkihiroSuda** [comment]
@stevvooe updated

**Comment by vdemeester** [comment]
ping @stevvooe 🙏 👼

**Comment by AkihiroSuda** [comment]
@stevvooe PTAL?

**Review by vdemeester — APPROVED** [review]
LGTM :cactus:

**Review by thaJeztah — APPROVED** [review]
LGTM (@stevvooe gave his lgtm on slack)

**Pull request merged**